tests/validators/test_dataclasses.py

```diff
@@ -1190,3 +1191,34 @@ def test_custom_dataclass_names():
         },
         {'input': 123, 'loc': ('foo', 'none'), 'msg': 'Input should be None', 'type': 'none_required'},
     ]
+
+
+@pytest.mark.skipif(sys.version_info < (3, 10), reason='slots are only supported for dataclasses in Python > 3.10')
+def test_slots() -> None:
+    kwargs = {'slots': True}
```

Comment:
@hramezani please make this test pass. We should also add a test for a function validator stuck in between the `DataclassSchema` and `DataclassArgsSchema` for both `before` and `after` variants. See for example `test_dataclass_field_before_validator`.